service/src/main/java/cz/muni/fi/pv254/parsing/CbAlgorithm.java

```diff
@@ -0,0 +1,198 @@
+            return gameA;
+        }
+
+        public void setGameA(GameDTO gameA) {
+            this.gameA = gameA;
+        }
+
+        public GameDTO getGameB() {
+            return gameB;
+        }
+
+        public void setGameB(GameDTO gameB) {
+            this.gameB = gameB;
+        }
+
+        public double getScoreAB() {
+            return scoreAB;
+        }
+
+        public void setScoreAB(double scoreAB) {
+            this.scoreAB = scoreAB;
+        }
+
+
+    }
+
+
+    @Autowired
+    private GameFacade gameFacade;
+    @Autowired
+    private RecommendationFacade recommendationFacade;
+    @Autowired
+    private UserFacade userFacade;
+    @Autowired
+    private GenreFacade genreFacade;
+
+    /**
+     *  https://www.linkedin.com/pulse/content-based-recommender-engine-under-hood-venkat-raman
+     *  https://ieeexplore.ieee.org/stamp/stamp.jsp?arnumber=6838200
+
+     * use this altgorithm  https://medium.com/@adriensieg/text-similarities-da019229c894?fbclid=IwAR1bxMdKdAWh7mAWTnAi1dQQV_q5v69lqudwUsypni3773dj-4W7sDscYf4
+     * hraA----hodnotena.....hraB---nehodnotena......pre kazde slovo z hry A najdem slovo v hreB...ak tam je vynasobim ich count a dalej scitavam cv citateli.......menovatel vsetky slova
+     * @param userDTO
+     * @return
+     */
+    private Set<GameDTO> recommendationByWord(UserDTO userDTO){
+
+        Set<GameDTO> userGames= new HashSet<>();
+        Set<GameDTO> VoteUserGames= new HashSet<>();
+        Set<GameDTO> bestScore = new HashSet<>();
+        Map<String, Double> abScore = new HashMap<>();
+        for(RecommendationDTO reDTO: userDTO.getRecommendations()){
+                if(reDTO.isVotedUp()){
+                    VoteUserGames.add(reDTO.getGame());
+                }
+                userGames.add(reDTO.getGame());
+
+        }
+        List<GameDTO> notRecUserGames =gameFacade.findAll();
+        for(GameDTO game: userGames){
+            notRecUserGames.remove(game);
+        }
+        //hodnotena hra
+        for(GameDTO Votegame: VoteUserGames){
+            Double ScoreSumAB=0D;
+            for(GameDTO NotVotegame: notRecUserGames){
+                Set<WordDTO> voteWords= Votegame.getWords();
+                Set<WordDTO> NotVoteWords= NotVotegame.getWords();
+                for(WordDTO Voteword: voteWords){
+                    for(WordDTO NotVoteword: NotVoteWords){
+                        if(Voteword.getWord().equals(NotVoteword.getWord())){
+                            ScoreSumAB=ScoreSumAB+(Voteword.getCount()*NotVoteword.getCount());
+                        }
+                    }
+                }
+                CBentityAB cBentityAB=new CBentityAB();
+                cBentityAB.setGameA(Votegame);
+                //nehodnotene
+                cBentityAB.setGameB(NotVotegame);
+                cBentityAB.setScoreAB(ScoreSumAB/(Math.sqrt(scoreA(Votegame)*scoreA(NotVotegame))*Math.sqrt(scoreA(Votegame)*scoreA(Votegame))));
```

Comment:
to prve nema byt nahodou notvotegame v oboch scoreA() ? ked tak sprav score A nech to rovno pocita druhu mocninu a tak sumu, nie som si isty ci je to mozne spravit naopak, najprv sumu tak na druhu; teda ci sum(xi^2) == sum(xi)^2, mam pocit ze to neni to iste